ui/src/app/domains/views/DomainDetails/ResourceRecords/ResourceRecords.tsx

```diff
@@ -14,47 +26,121 @@ const ResourceRecords = ({ id }: Props): JSX.Element | null => {
     domainsSelectors.getById(state, Number(id))
   );
 
+  const isAdmin = useSelector(authSelectors.isAdmin);
+
   if (!domain || !domain.rrsets || domain.rrsets.length === 0) {
     return null;
   }
 
   const headers = [
     {
       content: "Name",
+      sortKey: "name",
     },
     {
       content: "Type",
+      sortKey: "type",
     },
     {
       content: "TTL",
+      sortKey: "ttl",
     },
     {
       content: "Data",
+      sortKey: "data",
     },
     {
       content: "Actions",
+      className: "u-align--right",
     },
   ];
 
   const rows = domain.rrsets.map((resource) => {
+    const nodeType: NodeType = resource.node_type;
+    let nameCell: JSX.Element;
+
+    // We can't edit records that don't have a dnsresource_id.
+    // (If the row doesn't have one, it has probably been automatically
+    // generated by means of a deployed node, or some other reason.)
+    const isAutogenerated = !resource.dnsresource_id;
+    const canEdit = !isAutogenerated && isAdmin;
+
+    switch (nodeType) {
+      case NodeType.MACHINE:
+        nameCell = (
+          <Link to={machineURLs.machine.index({ id: resource.system_id })}>
```

Comment:
The url function requires a string, but `resource.system_id` can be `string | null` (if using the type suggestions above). Intuitively we know that if `resource.node_type` is not `null`, then `resource.system_id` will also not be `null`, but the TS compiler doesn't know that. So what we actually want to do is check:
``` typescript
if (resource.node_type === NodeType.MACHINE && resource.system_id !== null) {
  // do the thing
}
```